lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -606,6 +607,23 @@ export default function getAPIImplementation(): APIInterface {
             // with the same email are verified. The function below checks for the email verification
             // so we want to call it only once this is up to date,
 
+            // List down the edge cases for IP address based anomaly detection
+            // Implement IP address based AnomalyDetection
+
+            if (AnomalyDetection.getInstance() !== undefined) {
```

Comment:
this also needs to go into the session refreshing api as we had discussed. Over there though, we don't have a status like `SIGN_IN_NOT_ALLOWED`, so instead, we just revoke the session. Please figure out how that is to be done and test it